#165305722  Articles CRUD

Amoswachira opened the pull request:
### What does this PR do?
- Allows a user to do CRUD operations on articles 
### What are the tasks to be completed?
-Post an article
-Get an article
-Delete an article
-Edit an article
### How can this be manually tested?
After cloning the repo:
```
$ git checkout ft-user-articles-crud-165305722
$npm install
$ npm npm run start:dev
```
### What are the relevant pivotal tracker stories?
[#165305722](https://www.pivotaltracker.com/n/projects/2326460/stories/165305722)

###  Screenshots

<img width="1440" alt="Screenshot 2019-06-06 at 19 37 47" src="https://user-images.githubusercontent.com/17140636/59050371-db752100-8892-11e9-83dc-bb83c05a4367.png">
<img width="1440" alt="Screenshot 2019-06-06 at 19 35 35" src="https://user-images.githubusercontent.com/17140636/59050407-edef5a80-8892-11e9-84ca-ed5bd2575c7a.png">
<img width="1440" alt="Screenshot 2019-06-06 at 19 35 09" src="https://user-images.githubusercontent.com/17140636/59050426-f6479580-8892-11e9-80a8-7118eb30ecd1.png">

<img width="1440" alt="Screenshot 2019-06-13 at 12 02 47" src="https://user-images.githubusercontent.com/17140636/59419550-cd178f80-8dd3-11e9-98e3-be98e58c7996.png">

<img width="1440" alt="Screenshot 2019-06-13 at 12 02 55" src="https://user-images.githubusercontent.com/17140636/59419435-980b3d00-8dd3-11e9-99d2-dc5b480de30e.png">


<img width="1440" alt="Screenshot 2019-06-11 at 15 22 50" src="https://user-images.githubusercontent.com/7009215/59329616-f82cb100-8cf7-11e9-9186-f6da94383366.png">

Ogutu-Brian commented on .babelrc:
```diff
@@ -1,3 +1,11 @@
+            "@babel/plugin-proposal-class-properties"
+        ]
+    ]
+}
```
Hello @Amoswachira and @Bakley ,
Please act on the alert raised by GitHub on `No newline at end of file`

Amoswachira replied:
@Ogutu-Brian  I will fix that

Ogutu-Brian commented on jest.config.js:
```diff
@@ -11,7 +11,6 @@ module.exports = {
   verbose: false,
   moduleNameMapper: {
-    '^.+\\.(css|less|scss)$': 'babel-jest',
-    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$': '<rootDir>/__mocks__/fileMock.js',
```
Hello @Amoswachira and @Bakley ,
When the regex is removed, static files like images would throw errors during testing. 
For it to work, you should restore the `__mocks__` directory and the `__mocks__/fileMock.js` file that you have deleted

Ogutu-Brian commented on package.json:
```diff
@@ -34,42 +35,47 @@
   "dependencies": {
     "@babel/core": "^7.4.5",
     "@babel/polyfill": "^7.4.4",
-    "@fortawesome/fontawesome-svg-core": "^1.2.18",
```
Hello @Amoswachira and @Bakley ,
Why have you removed the font-awesome modules?
Since we are using them in our application, I think they should not be done away with.

Amoswachira replied:
@Ogutu-Brian  there is another font awesome CND link include in the index.html so the font awesome icons will still load

Ogutu-Brian replied:
Okay

Ogutu-Brian commented on src/views/Articles/Posts.jsx:
```diff
@@ -0,0 +1,51 @@
+      <div className="col-md-12">
+        <PostForm />
+        {postContent}
+      </div>
```
Hello @Amoswachira and @Bakley ,
Please do away with the empty lines ie. from 31 to 37, it makes the code unnecessarily long.

Amoswachira replied:
@Ogutu-Brian it is still a work in progress the final push will fix all the errors

Ogutu-Brian commented on src/views/Article/Spinner.jsx:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import spinner from './spinner.gif';
+
+const Spinner = () => {
```
Hello @Amoswachira ,
You had already defined a similar component in the components..Please reuse it to avoid code duplication
You should also treat views as only those components that connect to the store

Amoswachira replied:
@Ogutu-Brian I will make that spinner reusable thanks

Ogutu-Brian commented on package.json:
```diff
@@ -50,26 +54,31 @@
     "@fortawesome/react-fontawesome": "^0.1.4",
     "acorn": "^6.1.1",
-    "axios": "^0.19.0",
+    "axios": "^0.18.1",
```
Hello @Amoswachira ,
Why have you chosen to use a lower version of axios?

Amoswachira committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

Amoswachira committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

Amoswachira committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

Amoswachira committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

Amoswachira committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

a-braham committed:
165305722-ft(Articles):Articles CRUD
 -Articles CRUD
 -Add tests
 -Implement feedback
 -Fix bugs
[Finishes #165305722]

a-braham committed:
166604435-ch(Logout): User logout or redirect to login if not logged in (#25)

- Fix logged out user in api
- Added logout to clear localstorage
- Handled redirect to home page for protected routes

[Finishes #166604435]

IssaIan committed:
ft(Homepage): Add top top rated articles and popular articles

- Add static cards to homepage

[finishes #165305722]

IssaIan committed:
ft(Articles):Articles CRUD

- Implement articles CRUD
- Add tests
- Add articles to homepage
- Add static top rated articles and popular articles to homepage

[finishes #165305722]

IssaIan reviewed (approved):
LGTM

matthenge reviewed (approved):
LGTM
